### DISCUSSION TITLE
Preserve KePub format for kobo books

### DISCUSSION THREAD

**Discussion by Lachette** [apprenticeharper/DeDRM_tools#1845] [Ideas]
Buying books from the Kobo store, they are downloaded to the kobo ereader as kepub format, is there a way for dedrm them and preserving the kepub format? Specifically, the manga I purchase from the kobo store, looses its formatting and page display properties when converted over to epub after using dedrm.

**Comment by ElleKayEm** [comment]
Use Obok DeDRM to decrypt books from your Kobo ereader or Kobo Desktop program.

**Reply by Lachette** [reply]
I use the Obok DeDRM button on the top of calibre to decrypt the books. but after the process, the books that show up in the library are in epub format. is there a way to change how obok dedrm outputs the format?

**Reply by ElleKayEm** [reply]
They aren't changed, but non-Kobo devices/apps wouldn't recognize the kepub extension.  Kepubs are just epubs with some extra tags that only Kobo devices/apps will make use of.  Other epub readers will ignore.

**Reply by Lachette** [reply]
I understand that non-Kobo devices wouldn't recognize it, but considering in this case that they are originally coming off of a Kobo device, it would have been nice to preserve the original file format rather than going thru the trouble of removing DRM and then reconverting them back to kepub for use in the Kobo it came from - hence the suggestion.

**Reply by ElleKayEm** [reply]
I would think changing the extension would do it.  No conversion necessary.

**Reply by Lachette** [reply]
Sorry it took a while to reply. 

With regards to just changing the extension, my Kobo device will not read that file properly.
So my workflow is as follows:
1. Download ebook to kobo
2. DeDRM to remove drm
3. pull the file from calibre library
4. covert back to kePub using a different tool
5. import back to library in calibre.

One could see that with a simple option of preserving the kePub format during DRM removal would save alot of effort and hassle.

**Comment by ElleKayEm** [comment]
Development for DeDRM and Obok has moved to here:  https://github.com/noDRM/DeDRM_tools
